pkg/cloud/services/ec2/instances.go

```diff
@@ -259,6 +222,81 @@ func (s *Service) CreateInstance(scope *scope.MachineScope, userData []byte) (*i
 	return out, nil
 }
 
+// findSubnet attempts to retrieve a subnet ID in the following order:
+// - subnetID specified in machine configuration,
+// - subnet based on filters in machine configuration
+// - subnet based on the availability zone specified,
+// - default to the first private subnet available.
+func (s *Service) findSubnet(scope *scope.MachineScope) (string, error) {
+	// Check Machine.Spec.FailureDomain first as it's used by KubeadmControlPlane to spread machines across failure domains.
+	failureDomain := scope.Machine.Spec.FailureDomain
+	if failureDomain == nil {
+		failureDomain = scope.AWSMachine.Spec.FailureDomain
+	}
+
+	switch {
+	case scope.AWSMachine.Spec.Subnet != nil && scope.AWSMachine.Spec.Subnet.ID != nil:
+		return *scope.AWSMachine.Spec.Subnet.ID, nil
```

Comment:
I'm wondering if it would make sense to add a validation step here that the specified Subnet.ID does not conflict with the failureDomain if set...  Not something that needs to be solved as part of this PR, though.

Reply:
Yeah it definitely makes sense to validate that. Created #1838